src/dispatching/dispatcher.rs

```diff
@@ -75,17 +75,67 @@ where
         Self { dependencies, ..self }
     }
 
+    /// Specifies size of the queue for workers.
+    ///
+    /// By default it's 64.
+    #[must_use]
+    pub fn worker_queue_size(self, size: usize) -> Self {
+        Self { worker_queue_size: size, ..self }
+    }
+
+    /// Specifies the distribution function that decides how updates are grouped
+    /// before execution.
+    pub fn distribution_function<K>(
+        self,
+        f: fn(&Update) -> Option<K>,
+    ) -> DispatcherBuilder<R, Err, K>
+    where
+        K: Hash + Eq,
+    {
+        let Self {
+            bot,
+            dependencies,
+            handler,
+            default_handler,
+            error_handler,
+            distribution_f: _,
+            worker_queue_size,
+        } = self;
+
+        DispatcherBuilder {
+            bot,
+            dependencies,
+            handler,
+            default_handler,
+            error_handler,
+            distribution_f: f,
+            worker_queue_size,
+        }
```

Comment:
Why not `self.distribution_f = f; self`?

Reply:
because `self.distribution_f` is `fn(&Update) -> Option<Key>`, and `f` is `fn(&Update) -> Option<K>`, types are different